tests/rules/cross_function_tainting/test_csharp_method_overloading.cs

```diff
@@ -0,0 +1,33 @@
+class C
+{
+    string data;
+
+    void Process(int first, string second)
+    {
+        this.data = second;
+    }
+    void Process(int number)
+    {
+        this.data = number;
+    }
+
+    void Process(string input)
+    {
+        this.data = input;
+    }
+
+    static void Main()
+    {
+        C c = new C();
+        string tainted = source();
+        c.Process(tainted);
+        // OK: csharp_method_overload not found, double method with this arity
```

Comment:
How about without `--taint-intrafile`?

Is the default for `c` to not become tainted in such cases, by side-effect for example?